python/tank/bootstrap/manager.py

```diff
@@ -680,7 +689,7 @@ def _bootstrap_sgtk(self, engine_name, entity, progress_callback=None):
         self._report_progress(progress_callback, self._STARTING_TOOLKIT_RATE, "Starting up Toolkit...")
         tk = config.get_tk_instance(self._sg_user)
 
-        if not isinstance(config, InstalledConfiguration):
+        if not config.has_local_bundle_cache:
```

Comment:
do you think it could make more sense to move the `cache_apps()` method to the `Configuration` class rather than adding this new property? I think that could potentially end up being cleaner. Feels like `cache_apps()` could belong nicely as a method on the `Config` object. And add it to the base class and override it in the `InstalledConfiguration` with a nop method.